### PR TITLE
Conf UI

### PR DESCRIPTION
**Pull request by krzysztofwolski**
- define configuration shape using JSONSchema
- automatically generated forms with validation
- endpoint factory

The purpose of this PR is to explore automatic form generation, bundled with keeping the data in the SettingsManager, for fast app bootstrapping.

### CODE REVIEW

File: src/lib/confSchema.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import type { FromSchema } from "json-schema-to-ts";
+import { JSONConfiguration } from "./JSONConfiguration";
+
+export const exampleSchema = {
```
Comment (lkostrowski) [inline src/lib/confSchema.ts:4]: can we make it type safe?

**Review by lkostrowski — COMMENTED** [review]
Maybe we should try to figure out public API first and later focus on implementation?
1. Will it be a separate package? (eg `@saleor/app-form-builder`)
2. What is the usage/public API

File: package.json [hunk 2]
```diff
@@ -17,11 +17,17 @@
     "@material-ui/core": "^4.12.4",
     "@material-ui/icons": "^4.11.3",
     "@material-ui/lab": "4.0.0-alpha.61",
-    "@saleor/app-sdk": "0.18.0",
+    "@rjsf/core": "5.0.0-beta.13",
```
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply package.json:20]: It uses an updated version of the JSON schema and has a few fixes. The package is actively maintained and working on finalizing the beta soon

File: src/lib/confSchema.ts (hunk 1 above)
Comment (krzysztofwolski) [reply src/lib/confSchema.ts:4]: Yeah, that is the issue I mentioned ealier:
- this object can be typesafe by using `import { RJSFSchema } from "@rjsf/utils";` type 
- but using this type breaks `json-schema-to-ts`, which can produce a type of defined schema

**Pull request closed without merging**